unit-tests/log/test-invalid-enum.py

```diff
@@ -0,0 +1,26 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+from rspy import log, test
+import pyrealsense2 as rs
+import common
+
+
+#############################################################################################
+#
+test.start( f"Logging with invalid enum" )
+
+try:
+    rs.log_to_callback( rs.log_severity.debug, common.message_counter )
+    test.check_equal( common.n_messages, 0 )
+    # Following will throw a recoverable_exception, which will issue a log by itself!
+    test.check_throws( lambda: rs.log( rs.log_severity( 10 ), "10" ), RuntimeError, 'invalid enum value for argument "severity"' )
```

Comment:
This cast, and the one in the next line, work on my system: I get a `RuntimeError` from the C++.
But if you look, our LibCI somehow fails with a `TypeError` in the Python -- the cast cannot be done in earlier Python versions, it seems...
Any preference on how to deal with this?

Reply:
`try` `catch` and pass on catch?

Reply:
No, the whole point is to check the C++ error. Saying "this Python code may fail" invalidates the whole test.
I traced it to this:
Python **3.6** fails for the `-1` value, but works for, e.g., `10` — both are invalid enum values. It gives:
```
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. pyrealsense2.pyrealsense2.log_severity(value: int)
Invoked with: -1
```
Somehow `-1` isn't an int?
Python **3.9** works for both, even with negative values. So they fixed something it seems...
I'll comment out the -1 line and add explanation...

Reply:
OK

Reply:
Depends how anal you want to be :)
If we were lax like that, I would not have caught the problem with the Python. And that would have been a bug...
I think it's currently generic enough and specific enough, at the same time.

Reply:
How is it generic enough?
I know this function should throw but I am not sure what exception type will the python get.
You make me run it, verify, and they specifically write it in the test, sounds like a punishment :)
I just wanted to make sure it throws..

But that's not related to this PR

Reply:
I think it should be explicit :)